plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -552,17 +552,16 @@ public static LocationProvider getLocationProvider(SchemaTableName schemaTableNa
     public static Schema schemaFromMetadata(List<ColumnMetadata> columns)
     {
         List<NestedField> icebergColumns = new ArrayList<>();
+        AtomicInteger nextFieldId = new AtomicInteger(columns.size() + 1);
         for (ColumnMetadata column : columns) {
             if (!column.isHidden()) {
                 int index = icebergColumns.size();
```

Comment:
Use `nextFieldId.getAndIncrement()` here

Reply:
But as I know, create a new Iceberg table, numbering sequence from parent to child, if we hava a table schema is `a int, b array<int>, c map<string,string>`, schema id is `a -> 1`, `b -> 2`, `c -> 3`, `b.element -> 4`, `c.key -> 5`, `c.value -> 6`. I don't know if it's necessary.

At the same time, I did not consider the problem of hidden fields. If I understand correctly, the hidden field should be a very large id, so it is necessary to exclude the number of hidden fields for id binding.

Reply:
I agree that we should just use `nextFieldId.getAndIncrement()`. IMO the spec doesn't say whether the top-level ids should be incrementing. cc @findepi